app_dart/lib/src/service/luci_build_service.dart

```diff
@@ -539,89 +551,63 @@ class LuciBuildService {
-      'commit/git/$commitSha',
-      'commit/gitiles/chromium.googlesource.com/external/github.com/flutter/$repo/+/$commitSha',
-    ];
-    tags['user_agent'] = <String>['luci-scheduler'];
-    // Tag `scheduler_job_id` is needed when calling buildbucket search build API.
-    tags['scheduler_job_id'] = <String>['flutter/$builderName'];
-    localProperties['git_ref'] = commitSha;
-    return buildBucketClient.scheduleBuild(ScheduleBuildRequest(
-      builderId: BuilderId(
-        project: 'flutter',
-        bucket: bucket,
-        builder: builderName,
-      ),
-      gitilesCommit: GitilesCommit(
-        project: 'external/github.com/flutter/$repo',
-        host: 'chromium.googlesource.com',
-        ref: 'refs/heads/$branch',
-        hash: commitSha,
-      ),
-      tags: tags,
-      properties: localProperties,
-      // Run manual retries with higher priority to ensure tasks that can
-      // potentially open the tree are not wating for ~30 mins in the queue.
-      priority: 29,
-    ));
-  }
-
   /// Check to auto-rerun TOT test failures.
+  ///
+  /// A builder will be retried if:
+  ///   1. It has been tried below the max retry limit
+  ///   2. It is for the tip of tree
+  ///   3. The last known status is not green
+  ///   4. [ignoreChecks] is false. This allows manual reruns to bypass the Cocoon state.
   Future<bool> checkRerunBuilder({
     required Commit commit,
-    required LuciTask luciTask,
-    required int retries,
-    String repo = 'flutter',
-    DatastoreService? datastore,
-    bool? isFlaky,
+    required Target target,
+    required Task task,
+    required DatastoreService datastore,
+    Map<String, List<String>>? tags,
+    bool ignoreChecks = false,
   }) async {
-    if (await _shouldRerunBuilder(luciTask, retries, commit, datastore)) {
-      log.info('Rerun builder: ${luciTask.builderName} for commit ${commit.sha}');
-      final Map<String, List<String>> tags = <String, List<String>>{
-        'triggered_by': <String>['cocoon'],
-        'trigger_type': <String>['retry'],
-      };
-      await reschedulePostsubmitBuild(
-        commitSha: commit.sha!,
-        builderName: luciTask.builderName,
-        repo: repo,
-        tags: tags,
-        properties: repo == 'engine' ? Config.engineDefaultProperties : const <String, dynamic>{},
-        bucket: isFlaky ?? false ? 'staging' : 'prod',
-      );
-      return true;
+    if (ignoreChecks == false && await _shouldRerunBuilder(task, commit, datastore) == false) {
+      return false;
     }
-    return false;
+    log.info('Rerun builder: ${target.value.name} for commit ${commit.sha}');
+    tags ??= <String, List<String>>{};
+    tags['trigger_type'] = <String>['retry'];
+
+    final BatchRequest request = BatchRequest(
+      requests: <Request>[
+        Request(
+          scheduleBuild: _createPostsubmitScheduleBuild(
+            commit: commit,
+            target: target,
+            task: task,
+            priority: 29,
```

Comment:
Nit: add a doc.

Reply:
I moved this into a const variable (along with the other priorities)